frontend/src/app/shared/components/datepicker/helpers/date-modal.helpers.ts

```diff
@@ -74,8 +74,20 @@ export function keepCurrentlyActiveMonth(datePicker:DatePicker, currentMonth:num
 }
 
 export function setDates(dates:DateOption|DateOption[], datePicker:DatePicker, enforceDate?:Date):void {
-  const { currentMonth } = datePicker.datepickerInstance;
-  const { currentYear } = datePicker.datepickerInstance;
+  const { currentMonth, currentYear, selectedDates } = datePicker.datepickerInstance;
+
+  const [newStart, newEnd] = _.castArray(dates);
+  const [selectedStart, selectedEnd] = selectedDates;
+
+  // In case the new times match the current times, do not try to update
+  // the current selected months (Regression #46488)
+  if (selectedDates.length > 0
+    && newStart !== ''
+    && (newStart as Date).getTime() === selectedStart.getTime()
+    && (newEnd === selectedEnd || !!newEnd === !!selectedEnd)) {
```

Comment:
Not sure, I read this expression correctly.

But, if I have a `newStart`, `newEnd`, `selectedStart`, and `selectedEnd`. start dates are the same, but end dates are not, this expression is truthy and returns without setting the dates in the datepicker are not set in line 91. 

This happens, because both end dates are truthy and thus line 87 is truthy. Am I wrong?

Reply:
yes, correct. I've improved the check

Reply:
yes, you're right. I've improved the check